packages/app/src/cli/models/app/loader.test.ts

```diff
@@ -2377,6 +2378,31 @@ wrong = "property"
     expect(reloadedApp.usesWorkspaces).toBe(app.usesWorkspaces)
   })
 
+  test('does not call app.generateExtensionTypes by default', async () => {
+    // Given
+    await writeConfig(appConfiguration)
+    const generateTypesSpy = vi.spyOn(App.prototype, 'generateExtensionTypes')
+    // When
+    await loadTestingApp()
+    // Then
+    expect(generateTypesSpy).not.toHaveBeenCalled()
+    generateTypesSpy.mockRestore()
+  })
+
+  test('call app.generateExtensionTypes when REMOTE_DOM_EXPERIMENT is true', async () => {
+    process.env.REMOTE_DOM_EXPERIMENT = 'true'
```

Comment:
We prefer not to modify the real `process.env` in a test. Because the process might crash before you can clean it up.

Instead, we should inject the `env` to the loader so it can be easily replaced in a test:
This can be easily solved with few changes:
- modify `AppLoaderConstructorArgs` to accept an optional `env?: NodeJS.ProcessEnv`
- Add a new property to the loader and set it in the constructor: `private readonly env: NodeJS.ProcessEnv` & `this.env = env ?? process.env`
- use `this.env.REMOTE_DOM_EXPERIMENT` inside the loader.
- In the test, modify `loadTestingApp` to also accept an optional env `env?: NodeJS.ProcessEnv`
- When running your tests, inject the value you need when calling `loadTestingApp`